### PR TITLE
Housing Locator View for Unit Browsing

### LINKED ISSUE #21: Simple Unit Browsing: Housing Locator View
# Overview

## Frontend 

- Add the Pending Listings/All Listings toggle.
- Add the "+ Listing" floating action button
- Add the trash can icon to the Unit Card.
- Create the "Delete this Listing?" popup

[Figma](https://www.figma.com/file/06vEDN5gtlHlC5hxNXMb4g/USHS-Designs?node-id=337%3A9662&mode=dev)

### Units API

In `frontend/src/api/units.ts` create a `deleteUnit` function that calls the `DELETE /api/units/:id` route appropriately. You may need to add a `delete` function in `frontend/src/api/requests.ts`

### Functionality

When the "Pending Listings" is toggled:
1. Switch the header text from "Available Properties" to "Pending Listings".
3. Call `getAllUnits` and get only listings that have not been approved. Show that data on the page. 

..and vice versa when the "All Listings" is toggled.

When the trash can icon on a Unit Card is clicked, show the "Delete this Listing?" popup. If the "Delete" button is clicked, call the `deleteUnit` function from above. Then, refetch the list of units by calling `getAllUnits` and display the updated list of units.

### Delete Unit

Implement `DELETE /api/units` ([reference](https://docs.google.com/document/d/19EBAxzbCyPdT6RgELrGsq6A2RPeDt2OSi5DYbFWm03k/edit#bookmark=id.vxe5bt27ec6q)).

## Acceptance Criteria

- [x] Housing Locator-specific UI elements added to the Home page
- [x] Toggle pending units implementation
- [x] Delete Units API route function
- [x] `DELETE /api/units/:id` route

### LINKED ISSUE #22: Simple Unit Browsing [ Epic 🏔️ ]
## Overview

Let's implement the interface (and functionality) that will enable HLs and CMs to browse, search, and filter through units.

This is essentially the Home page of the application (`/`).

## Tasks
- [x] #20 
- [x] #21

### PR DESCRIPTION
**Pull request by Azhou2023**
## Tracking Info

Resolves #21 
Resolves #22 

## Changes

Added delete functionality in backend, added toggle button for pending/all listings, added delete icon, added floating listings button (all for housing locator specific view).

## Testing

Tested frontend components worked and interacted correctly, tested that delete route functioned correctly with dummy units.

## Confirmation of Change

Screenshot_14
Screenshot_15
Screenshot_16
Screenshot_19

### CODE REVIEW

File: frontend/src/api/requests.ts [hunk 1]
```diff
@@ -132,8 +132,16 @@ export async function put(
   return response;
 }
 
+//delete is a reserved word apparently
+export async function remove(url: string, headers: Record<string, string> = {}): Promise<Response> {
```
Comment (petabite) [inline frontend/src/api/requests.ts]: 😭 maybe call it `deleteRequest` for clarity?

**Review by petabite — APPROVED** [review]
LGTM 💯 Up to you whether you want to integrate the real unit data in another PR after #49 merges or just wait and include the changes in this PR.

**Review by petabite — APPROVED** [review]
🚢

**Review by YashRavipati1 — APPROVED** [review]
LGTM

**Pull request merged**